src/ucp/rndv/rndv_rtr.c

```diff
@@ -301,11 +301,9 @@ ucp_proto_rndv_rtr_mtype_abort(ucp_request_t *req, ucs_status_t status)
 
 static ucs_status_t ucp_proto_rndv_rtr_mtype_reset(ucp_request_t *req)
 {
-    ucs_mpool_put_inline(req->send.rndv.mdesc);
-    if (ucp_proto_rndv_request_is_ppln_frag(req)) {
-        req->status = UCS_ERR_CANCELED;
-        ucp_proto_rndv_ppln_recv_frag_clean(req);
-        return UCS_ERR_CANCELED;
+    if (req->flags & UCP_REQUEST_FLAG_PROTO_INITIALIZED) {
+        ucs_mpool_put_inline(req->send.rndv.mdesc);
+        req->send.rndv.mdesc = NULL;
```

Comment:
should not we propagate status to super recv request and reset it too?

Reply:
this is a reset function, why is status relevant here? Do you mean a user receive request? Not sure, it's not supposed to change afaiu

Reply:
ok